### ISSUE TITLE
Electra-StructAware replication

### ISSUE THREAD

**Issue by crarojasca** [causalNLP/logical-fallacy#3]
Hi,

This is Cristian Rojas from Monash University, we are replicating the results from the article https://doi.org/10.48550/arxiv.2202.13758 illustrated in Table 5 for the Electra models shared in the repository, following the script [logicedu](https://github.com/causalNLP/logical-fallacy/blob/main/codes_for_models/experiments_round2/logicedu.py), currently we can replicate the performance for the standard Electra Model by following the arguments:

`python logicedu.py  -t "google/electra-large-discriminator" -m "../../saved_models/electra-logic/" -ts 1 -ds 1 -nt T -mp base`

 - micro f1: 0.505771 
 - macro f1:0.475610 
 - precision: 0.512294 
 - recall: 0.803333 
 - exact match 0.333333

But we can't replicate the results for the struct-aware model yet, we are running the script in the next way:

`python logicedu.py -t "google/electra-large-discriminator"  -m "../../saved_models/electra-logic-structaware/"  -ts 2 -ds 2 -nt T -mp masked-logical-form`

 - micro f1: 0.411992 
 - macro f1:0.379006 
 - precision: 0.377926 
 - recall: 0.710000 
 - exact match 0.180000

Could check what are we missing in the configuration to replicate the results?

Cheers,
Cristian

**Comment by labhinav** [comment]
Hey Christian,

Can you try retraining the model from scratch (use https://huggingface.co/howey/electra-base-mnli as a base) and let us know what results you get?

**Comment by hieuchi911** [comment]
Hi, I'm also trying to replicate the results, and had the same results as Cristian's:

- Replicated results for Electra-logic:

  **Script**:
  `python logicedu.py  -t "google/electra-large-discriminator" -m "../../saved_models/electra-logic/" -ts 1 -ds 1 -nt T -mp base`
  
  **Results**:
  - micro f1: 0.505771
  - macro f1:0.475610
  - precision: 0.512294
  - recall: 0.803333
  - exact match 0.333333


- Replicated results for Electra-logic-structaware:

  **Script**:
  `python logicedu.py -t "google/electra-large-discriminator"  -m "../../saved_models/electra-logic-structaware/"  -ts 2 -ds 2 -nt T -mp masked-logical-form`
  
  **Results**:
  - micro f1: 0.417154
  - macro f1:0.383203
  - precision: 0.380667
  - recall: 0.713333
  - exact match 0.176667

- I also run the base model https://huggingface.co/howey/electra-base-mnli in `masked-logical-form` setting and got pretty bad results, which makes sense:

  **Script**:
  `python logicedu.py -t "google/electra-large-discriminator"  -m "howey/electra-base-mnli"  -ts 2 -ds 2 -nt T -mp masked-logical-form`
  
  **Results**:
  - micro f1: 0.092764
  - macro f1:0.079011
  - precision: 0.048636
  - recall: 0.083333
  - exact match 0.030000

**Comment by labhinav** [comment]
Hey,

I can see that your precision is higher than our reported results and the recall is lower. Can you try modifying the loss function to give a higher weightage to the positive class items? You can do this by setting the -w argument (try values between 10-20). Let us know if this works. We will add this to the list of recommended hyperparameters soon

**Comment by labhinav** [comment]
Closing due to no activity
